flow360/component/flow360_params/flow360_legacy.py

```diff
@@ -0,0 +1,621 @@
+        alias="maxForceJacUpdatePhysicalSteps"
+    )
+    linear_solver: Optional[LinearSolverLegacy] = pd.Field(
+        alias="linearSolver", default=LinearSolverLegacy()
+    )
+
+    def update_model(self) -> Flow360BaseModel:
+        model = {
+            "absoluteTolerance": self.absolute_tolerance,
+            "linearSolver": _try_update(self.linear_solver),
+            "equationEvalFrequency": self.equation_eval_frequency,
+        }
+
+        return HeatEquationSolver.parse_obj(model)
+
+
+class TransitionModelSolverLegacy(TransitionModelSolver, LegacyModel):
+    """:class:`TransitionModelSolverLegacy` class"""
+
+    reconstruction_gradient_limiter: Optional[pd.confloat(ge=0, le=2)] = pd.Field(
+        alias="reconstructionGradientLimiter"
+    )
+    CFL_multiplier: Optional[PositiveFloat] = pd.Field(alias="CFLMultiplier")
+    linear_iterations: Optional[PositiveInt] = pd.Field(alias="linearIterations")
+    randomizer: Optional[LinearIterationsRandomizer] = pd.Field()
+    linear_solver: Optional[LinearSolverLegacy] = pd.Field(
+        alias="linearSolver", default=LinearSolverLegacy()
+    )
+
+    def update_model(self) -> Flow360BaseModel:
+        model = {
+            "absoluteTolerance": self.absolute_tolerance,
+            "relativeTolerance": self.relative_tolerance,
+            "modelType": self.model_type,
+            "linearSolver": _try_update(self.linear_solver),
+            "updateJacobianFrequency": self.update_jacobian_frequency,
+            "equationEvalFrequency": self.equation_eval_frequency,
+            "maxForceJacUpdatePhysicalSteps": self.max_force_jac_update_physical_steps,
+            "orderOfAccuracy": self.order_of_accuracy,
+            "turbulenceIntensityPercent": self.turbulence_intensity_percent,
+            "Ncrit": self.N_crit,
+        }
+
+        return HeatEquationSolver.parse_obj(model)
+
+
+class BETDiskLegacy(BETDisk, LegacyModel):
+    """:class:`BETDiskLegacy` class"""
+
+    volume_name: Optional[str] = pd.Field(alias="volumeName")
+
+    def update_model(self):
+        model = {
+            "rotationDirectionRule": self.rotation_direction_rule,
+            "centerOfRotation": self.center_of_rotation,
+            "axisOfRotation": self.axis_of_rotation,
+            "numberOfBlades": self.number_of_blades,
+            "radius": self.radius,
+            "chordRef": self.chord_ref,
+            "thickness": self.thickness,
+            "nLoadingNodes": self.n_loading_nodes,
+            "bladeLineChord": self.blade_line_chord,
+            "initialBladeDirection": self.initial_blade_direction,
+            "tipGap": self.tip_gap,
+            "machNumbers": self.mach_numbers,
+            "reynoldsNumbers": self.reynolds_numbers,
+            "alphas": self.alphas,
+            "twists": self.twists,
+            "chords": self.chords,
+            "sectionalPolars": self.sectional_polars,
+            "sectionalRadiuses": self.sectional_radiuses,
+        }
```

Comment:
This was not done here mainly for consistency, for example in output classes we don't do it with dict() because this way all bool fields that are set to None are still set as a key in the dict. We would need to clean them. Additionally, I figured this way of doing it would be somewhat more clear for debugging what exactly gets created. If we want we can think of a more generic approach, sth like self.dict() but cull all keys not present in the non-legacy class. We would probably still need to handle some custom fields correctly.